packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -234,12 +234,14 @@ library LibProving {
         require(evidence.meta.id == target.id, "L1:height");
         require(evidence.prover != address(0), "L1:prover");
 
-        _checkMetadata({state: state, config: config, meta: target});
-        _validateHeaderForMetadata({
-            config: config,
-            header: evidence.header,
-            meta: evidence.meta
-        });
+        if (config.enableProofValidation) {
```

Comment:
OK, I will add the metadata changes for prove block in our utils function and then remove this.

Reply:
I think I am not fully understand how to convert some of our testnet blocks, that are not real testnet but a hardhat node, into a valid blockheader so that the blocks can be verified. If I manipulat the blockheader info to match the meta info, the block is unable to be verified because parentHash mismastch. I will continue to look into this.

Reply:
hmmm i also think its very hard for a hardhat block to pass the `_validateHeaderForMetadata` check, if we don't want to modify the metadata verification flow here, not sure but will creating a `TestLibProving.sol` and override the `_validateHeaderForMetadata` method be a better approach?

Reply:
I pushed up a new TestLibProving @davidtaikocha , it kind of sucks having to have a big `proveBlock` function since we cant override the private library methods that we want to disable (`_validateHeaderForMetadata`). That function would have to be public.